tools/AzureRM.BootStrapper/AzureRM.Bootstrapper.psm1

```diff
@@ -1055,3 +1123,126 @@ function Set-BootstrapRepo
 	param([string]$Repo)
 	$script:BootStrapRepo = $Repo
 }
+
+<#
+.ExternalHelp help\AzureRM.Bootstrapper-help.xml 
+#>
+function Set-AzureRmDefaultProfile
+{
+  [CmdletBinding(SupportsShouldProcess = $true)]
+  param()
+  DynamicParam
+  {
+    $params = New-Object -Type System.Management.Automation.RuntimeDefinedParameterDictionary
+    Add-ProfileParam $params
+    Add-ForceParam $params
+    Add-ScopeParam $params
+    return $params
+  }
+  PROCESS {
+    $armProfile = $PSBoundParameters.Profile
+    $Scope = $PSBoundParameters.Scope
+    $Force = $PSBoundParameters.Force
+
+    $defaultProfile = $Global:PSDefaultParameterValues["*-AzureRmProfile:Profile"]
+    if ($defaultProfile -ne $armProfile)
+    {
+      # Set DefaultProfile for this session
+      $Global:PSDefaultParameterValues["*-AzureRmProfile:Profile"]="$armProfile"
+
+      # Edit the profile content
+      $profileContent = @"
+          `$PSDefaultParameterValues["*-AzureRmProfile:Profile"]="$armProfile" `r`n
+"@
+
+      $scriptBlock = Get-ScriptBlock -armProfile $armProfile
+
+      $Global:PSDefaultParameterValues["Import-Module:RequiredVersion"]= $scriptBlock
+      $profileContent += @"
+          `$PSDefaultParameterValues["Import-Module:RequiredVersion"]={ $scriptBlock } `r`n
+"@
+      if ($PSCmdlet.ShouldProcess("$armProfile", "Set Default Profile")) 
+      {
+        if (($Force.IsPresent -or $PSCmdlet.ShouldContinue("Are you sure you would like to set $armProfile as Default Profile?", "Setting $armProfile as Default profile")))
+        {
+          # Check Profile existence and choose proper profile
+          $profilePath = Select-Profile -Scope $Scope
+
+          # Remove previous setting if exists
+          Write-Verbose "Updating default profile value to $armProfile"
+          Write-Debug "Removing previous setting if exists"
+          $RemoveSettingScriptBlock = {
+            (Get-Content -Path $profilePath -ErrorAction Stop) | Select-String -pattern 'AzureRmProfile:Profile' -notmatch | Set-Content -path  $profilePath -ErrorAction Stop
+            (Get-Content -Path $profilePath -ErrorAction Stop) | Select-String -pattern 'Import-Module:RequiredVersion' -notmatch | Set-Content -path  $profilePath -ErrorAction Stop
+            (Get-Content -Path $profilePath -ErrorAction Stop) | Select-String -pattern 'MyInvocation.Line.ToLower' -notmatch | Set-Content -path  $profilePath -ErrorAction Stop
+            (Get-Content -Path $profilePath -ErrorAction Stop) | Select-String -pattern '}' -notmatch | Set-Content -path  $profilePath -ErrorAction Stop
```

Comment:
I'm not sure if this is the right way to do it. This might remove '}' that was not intended to be removed. It would be great if you have any other mechanism for removing this scriptblock.